connection.go

```diff
@@ -1066,6 +1066,10 @@ func (conn *Connection) putFuture(fut *Future, req Request, streamId uint64) {
 	}
 	shard.bufmut.Unlock()
 
+	if firstWritten {
```

Comment:
How does this patch change the behavior? It seems to be related not only to watcher requests

Reply:
Now we release the future for an asynchronous request only after the request is queued to be sent. In general, this does not affect anything, it just makes the behavior more deterministic.

It is related only for async requests (IPROTO_WATCH, IPROTO_UNWATCH).